pgmpy/factors/CPD.py

```diff
@@ -107,27 +107,25 @@ def __init__(self, variable, variable_card, values,
 
         cardinality = [variable_card]
         if evidence_card is not None:
-            if not isinstance(evidence_card, (list, set, tuple)):
+            if not isinstance(evidence_card, (list, tuple)):
                 if isinstance(evidence_card, np.ndarray):
                     evidence_card = evidence_card.tolist()
                 elif isinstance(evidence_card, (int, float)):
                     evidence_card = [evidence_card]
                 else:
-                    raise TypeError("Evidence cardinality must be list, set, "
+                    raise TypeError("Evidence cardinality must be list, "
                                     "tuple, numpy ndarray or simply a number.")
-            self.evidence_card = evidence_card
             cardinality.extend(evidence_card[::-1])
```

Comment:
@ankurankan I am a little confused here. As I have already mentioned I didn't change these lines and removing `evidence` and `evidence_card` shouldn't have affected this as values being entered in the CPD are actually entered by calling its super class `Factor`. 
Also, although when using `print(cpd)` to print the cpd in my case gives weird output that I don't really understand why, below are some other things that I will need clarification upon.

Previous version, which is currently in use:

```python
>>> cpd = TabularCPD('A', 2, [[0.1, 0.2, 0.3, 0.4, 0.01, 0.02], [0.9, 0.8, 0.7, 0.6, 0.99, 0.98]], ['B', 'C'], [3, 2])
>>> cpd.values
array([[[ 0.1 ,  0.2 ,  0.3 ],
        [ 0.4 ,  0.01,  0.02]],

       [[ 0.9 ,  0.8 ,  0.7 ],
        [ 0.6 ,  0.99,  0.98]]])
>>> 
KeyboardInterrupt
>>> print(cpd)
+-----+-----+-----+-----+-----+-----+-----+
| B   | B_0 | B_0 | B_1 | B_1 | B_2 | B_2 |
+-----+-----+-----+-----+-----+-----+-----+
| C   | C_0 | C_1 | C_0 | C_1 | C_0 | C_1 |
+-----+-----+-----+-----+-----+-----+-----+
| A_0 | 0.1 | 0.2 | 0.3 | 0.4 | 0.0 | 0.0 |
+-----+-----+-----+-----+-----+-----+-----+
| A_1 | 0.9 | 0.8 | 0.7 | 0.6 | 0.9 | 0.9 |
+-----+-----+-----+-----+-----+-----+-----+
>>> cpd.variables
['A', 'C', 'B']
>>> cpd.cardinality
array([2, 2, 3])
```

My current version which reverses the list (as I believe was done previously too):

```python
>>> cpd = TabularCPD('A', 2, [[0.1, 0.2, 0.3, 0.4, 0.01, 0.02], [0.9, 0.8, 0.7, 0.6, 0.99, 0.98]], ['B', 'C'], [3, 2])
>>> cpd.values
array([[[ 0.1 ,  0.2 ,  0.3 ],
        [ 0.4 ,  0.01,  0.02]],

       [[ 0.9 ,  0.8 ,  0.7 ],
        [ 0.6 ,  0.99,  0.98]]])
>>> print(cpd)
+-----+-----+-----+-----+-----+-----+-----+
| C   | C_0 | C_0 | C_0 | C_1 | C_1 | C_1 |
+-----+-----+-----+-----+-----+-----+-----+
| B   | B_0 | B_1 | B_2 | B_0 | B_1 | B_2 |
+-----+-----+-----+-----+-----+-----+-----+
| A_0 | 0.1 | 0.2 | 0.3 | 0.4 | 0.0 | 0.0 |
+-----+-----+-----+-----+-----+-----+-----+
| A_1 | 0.9 | 0.8 | 0.7 | 0.6 | 0.9 | 0.9 |
+-----+-----+-----+-----+-----+-----+-----+
>>> cpd.variables
['A', 'C', 'B']
>>> cpd.cardinality
array([2, 2, 3])

```

If don't reverse the list, as suggested by you:

```python
>>> cpd = TabularCPD('A', 2, [[0.1, 0.2, 0.3, 0.4, 0.01, 0.02], [0.9, 0.8, 0.7, 0.6, 0.99, 0.98]], ['B', 'C'], [3, 2])
>>> cpd.values
array([[[ 0.1 ,  0.2 ],
        [ 0.3 ,  0.4 ],
        [ 0.01,  0.02]],

       [[ 0.9 ,  0.8 ],
        [ 0.7 ,  0.6 ],
        [ 0.99,  0.98]]])
>>> print(cpd)
+-----+-----+-----+-----+-----+-----+-----+
| B   | B_0 | B_0 | B_1 | B_1 | B_2 | B_2 |
+-----+-----+-----+-----+-----+-----+-----+
| C   | C_0 | C_1 | C_0 | C_1 | C_0 | C_1 |
+-----+-----+-----+-----+-----+-----+-----+
| A_0 | 0.1 | 0.2 | 0.3 | 0.4 | 0.0 | 0.0 |
+-----+-----+-----+-----+-----+-----+-----+
| A_1 | 0.9 | 0.8 | 0.7 | 0.6 | 0.9 | 0.9 |
+-----+-----+-----+-----+-----+-----+-----+
>>> cpd.variables
['A', 'B', 'C']
>>> cpd.cardinality
array([2, 3, 2])

```

Notice that in my case `cpd.values`, `cpd.variables`, `cpd.cardinality` match with the previous implementation but not the output with `print(cpd)`. 
So, I am reversing the lists as I think that values should be in the same format as they were previously for expected behavior of TabularCPD. 
I maybe completely wrong here. Let me know.